ports-of-call/portable_arrays.hpp

```diff
@@ -26,73 +26,142 @@
 
 #include "portability.hpp"
 #include <algorithm>
+#include <array>
 #include <assert.h>
 #include <cstddef> // size_t
 #include <cstring> // memset()
+#include <functional>
+#include <numeric>
+#include <type_traits>
 #include <utility> // swap()
 
+// maximum number of dimensions
+constexpr std::size_t MAXDIM = 6;
+// array type of dimensions/strides
+using narr = std::array<std::size_t, MAXDIM>;
+
+namespace detail {
+
+// set_value end-case
+template <std::size_t Ind, typename NX>
+constexpr void set_value(narr &ndat, NX value) {
+  ndat[Ind] = value;
```

Comment:
is `operator[]` of `std::array` marked  `constexpr`? I thought this was something that preventing more widespread usage.

Reply:
Sorry I intended to replace all these with appropriate `PORTABLE_` s but missed a few

Reply:
Oh, you probably meant something else. We can use a raw array, if `std::array` is messing with things

Reply:
We should check that it runs on device, but I *thought* `std::array` worked with `--expt-relaxed-constexpr` but I'm not sure

Reply:
@mauneyc-LANL you may be able to use `std::get<Ind>(arr) = value;`. But we really need to check and make sure it works on device.

Reply:
Using `--expt-relaxed-constexpr` this runs smoothly using `volta-x86` partition, `kokkos+cuda cuda_arch=70 +wrapper`, `cmake .. -DPORTS_OF_CALL_BUILD_TESTING -DPORTABILITY_STRATEGY_KOKKOS=ON`